tck/utils/key.ts

```diff
@@ -0,0 +1,104 @@
+import { PrivateKey, PublicKey, KeyList, Key } from "@hashgraph/sdk";
+import { proto } from "@hashgraph/proto";
+
+import { asn1DecodeStringDer } from "../utils/asn1-decoder";
+
+const getKeyFromString = (
+    keyString: string,
+): PublicKey | PrivateKey | Key | null => {
+    const decodedKey = asn1DecodeStringDer(keyString);
+
+    if (decodedKey.isKeyListHex) {
+        return getKeyFromKeyListHex(keyString);
+    }
+
+    try {
+        if (decodedKey.isPublicKey) {
+            if (decodedKey.keyTypes.includes("ecdsa")) {
+                return PublicKey.fromStringECDSA(keyString);
+            } else if (decodedKey.keyTypes.includes("ed25519")) {
+                return PublicKey.fromStringED25519(keyString);
+            }
+        } else {
+            if (decodedKey.keyTypes.includes("ecdsa")) {
+                return PrivateKey.fromStringECDSA(keyString);
+            } else if (decodedKey.keyTypes.includes("ed25519")) {
+                return PrivateKey.fromStringED25519(keyString);
+            }
+        }
+    } catch (error) {
+        console.error("Error parsing key: ", error);
+        return null;
+    }
+
+    return null;
+};
```

Comment:
Seems like this function has more than 1 responsibility. It handles both public keys, private keys, keylists. Maybe separate the parsing of these keys in different methods like `parsePublickey` `parsePrivatekey` and handle the errors in their respective functions. It's going to make the code more readable and reusable.